packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/CSharpGenTests.cs

```diff
@@ -45,25 +46,27 @@ public void TestGetConfigurationInputFilePath_DefaultPath()
         [Test]
         public void TestCSharpGen_ValidPlugin()
         {
-            // mock plugin
-            var mockPlugin = new Mock<CodeModelPlugin>(new GeneratorContext(Configuration.Load(MockHelpers.MocksFolder)))
-            {
-                CallBase = true
-            };
+            MockHelpers.LoadMockPlugin();
+            var csharpGen = new CSharpGen();
 
-            // mock type factory
-            var mockTypeFactory = new Mock<TypeFactory>()
-            {
-                CallBase = true
-            };
-
-            mockTypeFactory.Protected().Setup<CSharpType>("CreateCSharpTypeCore", ItExpr.IsAny<InputType>()).Returns(new CSharpType(typeof(IList<>)));
-            mockPlugin.SetupGet(p => p.TypeFactory).Returns(mockTypeFactory.Object);
-
-            var configFilePath = Path.Combine(MockHelpers.MocksFolder, "Configuration.json");
-            var csharpGen = new CSharpGen().ExecuteAsync();
+            Assert.DoesNotThrowAsync(async () => await csharpGen.ExecuteAsync());
+        }
 
-            Assert.IsNotNull(csharpGen);
```

Comment:
This was always not null before since it was just the created task which wasn't awaited.